pkg/wallet/wallet.go

```diff
@@ -519,18 +544,102 @@ func AddressIsUnique(address string) error {
 	return nil
 }
 
+// Helpers: public key
+
 // GetPupKey returns the public key of the wallet.
 func (wallet *Wallet) GetPupKey() string {
-	return PublicKeyPrefix + base58.CheckEncode(wallet.KeyPair.PublicKey, Base58Version)
+	return PublicKeyPrefix + base58.CheckEncode(RemovePubKeyVersion(wallet.KeyPair.PublicKey), PubKeyVersion)
+}
+
+// VersionedPubKey returns the public key of the wallet with a version byte prepended.
+func (wallet *Wallet) VersionedPubKey() []byte {
+	return addPubKeyVersion(wallet.KeyPair.PublicKey)
 }
 
+func addPubKeyVersion(pubKey []byte) []byte {
+	if len(pubKey) == ed25519.PublicKeySize {
+		return append([]byte{PubKeyVersion}, pubKey...)
+	}
+
+	return pubKey
+}
+
+// RemovePubKeyVersion removes the version byte of a public key if the public key has a version byte.
+func RemovePubKeyVersion(versionedPubKey []byte) []byte {
+	if len(versionedPubKey) == ed25519.PublicKeySize+1 {
```

Comment:
is there any expected cases where  this can be false ? if not you should just check the lenght prior to that and return an error if it is not the expected one

Reply:
you are right I reworked this part

Reply:
https://github.com/massalabs/station-massa-wallet/pull/530/commits/29a383abdc7c24b4e69f8b1130664fe267f7fee4